syntax = "proto3";

package GraphProto;

message Edge {
  uint64 from = 1;
  uint64 to = 2;
  double weight = 3;
}

message IncidenceList {
  repeated uint64 edge_ids = 1;
}

message DirectedWeightedGraph {
  repeated Edge edges = 1;
  repeated IncidenceList incidence_lists = 2;
}

message RouteInternalData {
  bool exists = 1;
  double weight = 2;
  bool has_prev_edge = 3;
  uint64 prev_edge = 4;
}

message RoutesInternalDataForOneVertex {
  repeated RouteInternalData routes_data_for_one_vertex = 1;
}

message Router {
  repeated RoutesInternalDataForOneVertex routes_data = 1;
}
